packages/site/pages/components/tooltip.js

```diff
@@ -4,6 +4,7 @@ import Icon from '@pluralsight/ps-design-system-icon/react'
 import React from 'react'
 import Theme from '@pluralsight/ps-design-system-theme/react'
 import Tooltip from '@pluralsight/ps-design-system-tooltip/react'
+import * as funcUtil from '@pluralsight/ps-design-system-util/func'
```

Comment:
Do we get better tree-shaking if we just import the function instead of the whole module with `* as`?  I'm not sure, but seems like I remember that... :)

Reply:
Great question, there are a few variables that would affect the final outcome of this:

* The first being consumer applications and their module bundler. Are they using a tool that is ES2015 module compliant and one that can perform tree-shaking (webpack, rollup, etc)?
* The second being how we build and publish our library. Do we include a `module` entry in the `package.json` of our published packages that points to a static ES2015 module compliant build?

As you probably know, libraries are often transpiled down to a dynamic module system like Common JS to support the widest possible execution target. We could look into publishing a second build that includes a static build that would take advantage of the ES2015 module system's killer feature: exporting more than a single thing! In our case here that would be a single `debounce` function, or in the future, other util functions that would be useful in isolation. This is in contrast to the Common JS single export workaround which involves exporting an object with keyed properties. This makes it impossible to determine what will not be used in the importing module, and thus prevents any optimization!

For now, I can easily rework the import/exports here to use only the function we care about. However it is a bit harder to say right now if end users in the browser will feel the effects of any optimization for the reasons I mention above!

Reply:
Oh man!  Great reminder.  I forgot about this.  And yes, we compile to cjs for our design system build.  And it's only using babel.  Yes, many more factors here to consider.  Fine for now.  That said, I'm going to go check out your other PR now.

Sorry for the slowness.  I've been out at a conf all week.